### PR TITLE
Enhance documentation generation using angular jsdoc

### PR DESCRIPTION
**Pull request by smbuthia**
# Description

Enhancement of documentation tooling using [angular-jsdoc](https://www.npmjs.com/package/angular-jsdoc)

Addition of documentation sources (admin, api, sentinel and webapp).

medic/medic#5548

# Code review items

- Readable: Concise, well named, follows the [style guide](https://github.com/medic/medic-docs/blob/master/development/style-guide.md), documented if necessary.
- Documented: Configuration and user documentation on [medic-docs](https://github.com/medic/medic-docs/)
- Tested: Unit and/or e2e where appropriate
- Internationalised: All user facing text
- Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in Changes.md.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -200,6 +200,8 @@ They live in [tests](tests). Run them with grunt: `grunt e2e`.
 
 To build reference documentation into a local folder `jsdoc-docs`: `grunt build-documentation`
 
+To add documentation, use these [tags](https://www.npmjs.com/package/angular-jsdoc#tags-available). See examples [here](https://www.npmjs.com/package/angular-jsdoc#example).
+
```
Comment (garethbowen) [inline README.md]: I think we should move this to the dev style guide: https://github.com/medic/medic-docs/blob/master/development/style-guide.md

Maybe also include a link to the jsdocs documentation too.

File: Gruntfile.js [hunk 2]
```diff
@@ -856,9 +856,21 @@ module.exports = function(grunt) {
...
+            'sentinel/src/**/*.js',
+            'webapp/src/js/*.js',
+            'webapp/src/js/**/*.js'
+          ],
           options: {
-              destination: 'jsdoc-docs'
+              destination: 'jsdoc-docs',
+              configure: 'node_modules/angular-jsdoc/common/conf.json',
+              template: 'node_modules/angular-jsdoc/angular-template',
+              readme: './README.md'
```
Comment (garethbowen) [inline Gruntfile.js]: I think we should build these as separate docs. If you're writing code for api it's not helpful to know what's in admin. It also means we can only use the angular jsdoc config and templates in admin and webapp.

File: Gruntfile.js [hunk 3]
```diff
@@ -856,9 +856,21 @@ module.exports = function(grunt) {
...
-          src: ['webapp/src/js/**/*.js', 'admin/src/js/**/*.js'],
+          src: [
+            'admin/src/js/*.js',
+            'admin/src/js/**/*.js',
+            'api/src/*.js',
+            'api/src/**/*.js',
+            'sentinel/src/*.js',
+            'sentinel/src/**/*.js',
+            'webapp/src/js/*.js',
+            'webapp/src/js/**/*.js'
```
Comment (garethbowen) [inline Gruntfile.js]: We should build the shared-libs too.

Comment (smbuthia) [reply Gruntfile.js]: Added this.

File: Gruntfile.js (hunk 2 above)
Comment (smbuthia) [reply Gruntfile.js]: Done. There might be a need to have a different template for the non-angular docs. I'm still looking into this.

File: README.md (hunk 1 above)
Comment (smbuthia) [reply README.md]: This is done. PR in medic-docs.

**Comment by smbuthia** [comment]
Tests failed the first time possibly because of https://github.com/medic/medic/issues/5829

**Pull request merged**